### ISSUE TITLE
[core] WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass

### ISSUE THREAD

**Issue by wener-tiobe** [pmd/pmd#4674]
**Affects PMD Version:** 

Tested on https://github.com/pmd/pmd/releases/download/pmd_releases%2F7.0.0-rc3/pmd-dist-7.0.0-rc3-bin.zip

**Description:**
When running CPD, I get a warning as depicted below.

**Sample demonstrating the issue:**
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/D:/tmp/test-pmd/pmd-bin-7.0.0-rc3/lib/groovy-2.4.21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[main] INFO net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand - Log level is at TRACE
[main] DEBUG net.sourceforge.pmd.cpd.CPD - Searching directory . for files
[main] DEBUG net.sourceforge.pmd.cpd.CPD - Tokenizing D:\10_Projects\Python\exp\.\__init__.py
[main] DEBUG net.sourceforge.pmd.cpd.CPD - Tokenizing D:\10_Projects\Python\exp\.\missing_comma.py
[main] DEBUG net.sourceforge.pmd.cpd.CPD - Running match algorithm on 2 files...
[main] DEBUG net.sourceforge.pmd.cpd.CPD - Finished: 0 duplicates found
```

**Steps to reproduce:**
1.  Download PMD 7.0.0. RC
2.  Invoke: `d:\tmp\test-pmd\pmd-bin-7.0.0-rc3\bin\pmd.bat cpd --minimum-tokens=100 --language python --debug --dir .`

**Running PMD through:** *[CLI]*

I use the following Java version:
```
java -version
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment Temurin-11.0.13+8 (build 11.0.13+8)
OpenJDK 64-Bit Server VM Temurin-11.0.13+8 (build 11.0.13+8, mixed mode)
```

**Comment by Scrates1** [comment]
I also meet this issue when using CPD：

**Affects PMD Version:**

7.0.0-rc3

**Description:**
When running CPD GUI, I get a warning as depicted below.

Sample demonstrating the issue:
```bash
...\pmd-bin-7.0.0-rc3\bin> ./pmd.bat cpd-gui
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/.../pmd-bin-7.0.0-rc3/lib/groovy-2.4.21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```
I use the following Java version:
corretto-11.0.18

**Comment by jorgesolebur** [comment]
Hello ! Same error in 7.0.0.-rc4
<img width="1342" alt="CleanShot 2023-10-06 at 00 12 53@2x" src="https://github.com/pmd/pmd/assets/31279773/ae6c92a9-c67a-4b92-9b63-dbd33e27777e">
